spy/backend/c/context.py

```diff
@@ -112,7 +98,7 @@ def c_function(self, name: str, w_func: W_ASTFunc) -> C_Function:
         for i, param in enumerate(w_functype.params):
             c_type = self.w2c(param.w_T)
             if param.kind == "simple":
-                c_param_name = check_c_preserve(funcdef.args[i].name)
+                c_param_name = str(C.Ident(funcdef.args[i].name))
```

Comment:
this calls to `str()` looks bad.
I think we can:
  1. move C.Ident to `context.py` and call it `C_Ident` (similarly to `C_Type`)
  2. modify `C_FuncParam.name` to be of type `C_Ident`

Generally speaking, you should never need an explicit call to `str(C_Ident(...))`, because `C_Ident.__str__` already does the job.